src/context.rs

```diff
@@ -63,6 +63,40 @@ impl Drop for RectangleList {
     }
 }
 
+impl fmt::Debug for RectangleList {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if self.ptr.is_null() {
+            write!(f, "RectangleList")
```

Comment:
Is this even valid? Judging from other code here it's not so we don't need to handle that here.